language/move-lang/stdlib/modules/gas_schedule.move

```diff
@@ -0,0 +1,47 @@
+address 0x0:
+
+// The gas schedule keeps two separate schedules for the gas:
+// * The instruction_schedule: This holds the gas for each bytecode instruction.
+// * The native_schedule: This holds the gas for used (per-byte operated over) for each native
+//   function.
+// A couple notes:
+// 1. In the case that an instruction is deleted from the bytecode, that part of the cost schedule
+//    still needs to remain the same; once a slot in the table is taken by an instruction, that is its
+//    slot for the rest of time (since that instruction could already exist in a module on-chain).
+// 2. The initialization of the module will publish the instruction table to the association
+//    address, and will preload the vector with the gas schedule for instructions. The VM will then
+//    load this into memory at the startup of each block.
+module GasSchedule {
+    use 0x0::Vector;
+    use 0x0::Transaction;
+
+    // The gas cost for each instruction is represented using two amounts;
+    // one for the cpu, and the other for storage.
+    struct Cost {
+      cpu: u64,
+      storage: u64,
+    }
+
+    resource struct T {
+        instruction_schedule: Vector::T<Cost>,
+        native_schedule: Vector::T<Cost>,
+    }
+
+    // Initialize the table under the association account
+    fun initialize(gas_schedule: T) {
```

Comment:
I would need to think more about this. But at least at the moment I don't want to make that switch until we make the switch away from the IR genesis.

Reply:
"doing this will require changing the rust-level genesis generation code." -> as in what it is right now?

Reply:
a note on the raw bytes: that would imply that we need and must specify a format for those bytes so that any VM, in any language can load them. It would also imply that a Move story for updates or reads would be impossible.
Not sure if that is a good thing or a bad thing ;) but figured I'd mention